### ISSUE TITLE
Add LoaderService and Component

### ISSUE THREAD

**Issue by M4R1KU** [outcobra/outstanding-cobra#144]
Add a loader service and a loader component to show on pageload and route changes.

**Comment by jmesserli** [comment]
Agreed that would be nice. Maybe based on NProgress? (rstacruz/nprogress)

**Comment by M4R1KU** [comment]
I would have used an overlay with the [md-progress-spinner](https://material.angular.io/components/component/progress-spinner) but we could also use the one you suggested.
